### ISSUE TITLE
[Bug]: isInternalNotification returning false for appboy_uninstall_tracking test push

### ISSUE THREAD

**Issue by Charbs09** [braze-inc/braze-swift-sdk#33]
### Platform

iOS

### Platform Version

16.1.1

### Braze SDK Version

5.8.1

### Xcode Version

Xcode 14.2

### Computer Processor

Apple (M1)

### Repro Rate

100% of the time

### Steps To Reproduce

1. Setup a test internal push per instructions here: https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/analytics/uninstall_tracking/#step-3-test-from-the-dashboard
2. In the didFinishLaunchingWithOptions function of the appDelegate, call this function 
```
@objc public static func clientShouldIgnoreLaunch(launchOptions: [UIApplication.LaunchOptionsKey: Any]) -> Bool {
        let pushDictionary = launchOptions[UIApplication.LaunchOptionsKey.remoteNotification] as? NSDictionary as? [AnyHashable : Any] ?? [:]
        
        return Braze.Notifications.isInternalNotification(pushDictionary)
    }
```

3. Send the push

### Expected Behavior

isInternalNotification should return true so that we are able to verify that uninstall tracking pushes will not run all of our app startup code

### Actual Incorrect Behavior

isInernalNotification returns false

**Comment by hokstuff** [comment]
Hi @Charbs09,

Can you share what is inside the `pushDictionary` object that is passed into `isInternalNotification` in your repro? Also, can you compare the contents of it with that of `userInfo` object when getting a push notification from uninstall tracking in `application(_:didReceiveRemoteNotification:fetchCompletionHandler:)`?

**Comment by Charbs09** [comment]
Hi @hokstuff,
The pushDictionary from the code snippet above looks like this:
`Printing description of pushDictionary:
▿ 6 elements
  ▿ 0 : 2 elements
    ▿ key : AnyHashable("testData1")
      - value : "testData1"
    - value : testData1
  ▿ 1 : 2 elements
    ▿ key : AnyHashable("ab_uri")
      - value : "ab_uri"
    - value : adobesparkpost://pricing
  ▿ 2 : 2 elements
    ▿ key : AnyHashable("aps")
      - value : "aps"
    ▿ value : 3 elements
      ▿ 0 : 2 elements
        - key : alert
        ▿ value : 2 elements
          ▿ 0 : 2 elements
            - key : title
            - value : Test Push Deep Link
          ▿ 1 : 2 elements
            - key : body
            - value : Tap this push notification to follow a deep link!
      ▿ 1 : 2 elements
        - key : interruption-level
        - value : active
      ▿ 2 : 2 elements
        - key : content-available
        - value : 1
  ▿ 3 : 2 elements
    ▿ key : AnyHashable("ab")
      - value : "ab"
    ▿ value : 1 element
      ▿ 0 : 2 elements
        - key : c
        - value : dGVzdF90ZXN0X2RpPTYzYmU5M2NhZjlkNzlmNzY4MjQ2ZWJiNzcwNTU1ZjRl
  ▿ 4 : 2 elements
    ▿ key : AnyHashable("testData2")
      - value : "testData2"
    - value : testData2
  ▿ 5 : 2 elements
    ▿ key : AnyHashable("appboy_uninstall_tracking")
      - value : "appboy_uninstall_tracking"
    - value : true`

I'll have to actually setup uninstall tracking on this new app group to compare against a real one. Will try to get to that this week.

What is interesting is that Braze.notifications.handleBackgroundNotification returns true (meaning yes it will handle) for this same test push, hence I was figuring there might be some issue in isInternalNotification specifically.

**Comment by hokstuff** [comment]
In your notification payload above, it contains extra fields which makes the entire notification no longer a silent push. By putting text inside the body of the silent push notification, the notification now has an `interruption-level` of `active` (copied from above):
```
  ▿ 2 : 2 elements
    ▿ key : AnyHashable("aps")
      - value : "aps"
    ▿ value : 3 elements
      ▿ 0 : 2 elements
        - key : alert
        ▿ value : 2 elements
          ▿ 0 : 2 elements
            - key : title
            - value : Test Push Deep Link
          ▿ 1 : 2 elements
            - key : body
            - value : Tap this push notification to follow a deep link!
      ▿ 1 : 2 elements
        - key : interruption-level
        - value : active
      ▿ 2 : 2 elements
        - key : content-available
        - value : 1
```

Can you try removing that extra part of the body from your campaign and try again? You may also consider trying using a real campaign instead of a test campaign if that doesn't work.

**Comment by Charbs09** [comment]
Ok, removing the content results in not seeing an actual push notification on my device, but isInternalNotification still returns false with this pushDictionary
```
▿ 3 elements
  ▿ 0 : 2 elements
    ▿ key : AnyHashable("ab")
      - value : "ab"
    ▿ value : 1 element
      ▿ 0 : 2 elements
        - key : c
        - value : dGVzdF90ZXN0X2RpPTYzYzA1ODQ3MWNkOGM4ODNkNjk0YzlmNzhkYjVjYjRh
  ▿ 1 : 2 elements
    ▿ key : AnyHashable("appboy_uninstall_tracking")
      - value : "appboy_uninstall_tracking"
    - value : true
  ▿ 2 : 2 elements
    ▿ key : AnyHashable("aps")
      - value : "aps"
    ▿ value : 2 elements
      ▿ 0 : 2 elements
        - key : interruption-level
        - value : active
      ▿ 1 : 2 elements
        - key : content-available
        - value : 1
```

Sending from a live campaign without any content has a similar package and isInternalNotification still returns false
```
▿ 3 elements
  ▿ 0 : 2 elements
    ▿ key : AnyHashable("appboy_uninstall_tracking")
      - value : "appboy_uninstall_tracking"
    - value : true
  ▿ 1 : 2 elements
    ▿ key : AnyHashable("ab")
      - value : "ab"
    ▿ value : 1 element
      ▿ 0 : 2 elements
        - key : c
        - value : NjNjMDU5N2Q1MTMzMjM1MjQ5NmE4OGMyXyRfZGk9NjNjMDU5YjcxNjE2MzM3ODMyNjUyOGZmOTA2OWJiMWMmbXY9NjNjMDU5N2Q1MTMzMjM1MjQ5NmE4OGM2JnBpPWNtcA==
  ▿ 2 : 2 elements
    ▿ key : AnyHashable("aps")
      - value : "aps"
    ▿ value : 2 elements
      ▿ 0 : 2 elements
        - key : interruption-level
        - value : active
      ▿ 1 : 2 elements
        - key : content-available
        - value : 1
 ```

**Comment by hokstuff** [comment]
Hi @Charbs09,

Thanks for testing this on your end! We have been able to repro what you're seeing and have identified a bug in the SDK, _specifically when testing by following_ [_these instructions_](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/analytics/uninstall_tracking/#step-2-checking-for-braze-background-push) (sending a custom KVP from the dashboard).

To clarify, the behavior when using Uninstall Tracking in production should function without issues and the value of `isInternalNotification` should be as expected.

We will follow up by extending the functionality of `isInternalNotification` to also correctly return the expected values when testing using a KVP in the dashboard.

**Comment by Charbs09** [comment]
Awesome! Thanks @hokstuff !

**Comment by cbowns** [comment]
@Charbs09 hi there! This should be fixed in our [5.10.0 release](https://github.com/braze-inc/braze-swift-sdk/releases/tag/5.10.0); let us know if that version resolves your issue.
